libtiledbvcf/src/write/writer.cc

```diff
@@ -1356,7 +1410,11 @@ void Writer::dataset_version(int32_t* version) const {
 }
 
 void Writer::finalize_query(std::unique_ptr<tiledb::Query> query) {
-  query->finalize();
+  if (remote_) {
+    query->submit_and_finalize();
+  } else {
+    query->finalize();
```

Comment:
We can call `submit_and_finalize` if the buffers are cleared. We should add a check to ensure the buffers are empty.

Reply:
All done here :tada:  I reran validation today for ingesting the full VCF dataset using instructions on [SC-24610](https://app.shortcut.com/tiledb-inc/story/24610/test-tile-aligned-write-caching-pr-3762-with-tiledb-vcf) and checksums are still good. If we can validate allele count or variant stats I can check those as well.

I had to be a bit more selective on when we zero buffers than I initially thought. If we set the query buffers with size of 0 after every submit and ingestion finds there's still more data to write, then we end up hitting [this](https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/query/query.cc#L1352) error in core if we attempt to re-set the previously zeroed buffers. So we only zero query buffers just before we finalize, and not after every submit.

Reply:
Thanks @shaunrd0, this is great work!